### ISSUE TITLE
Label name's link is split into two links by colon

### ISSUE THREAD

**Issue by yarakos95** [James-Yu/LaTeX-Workshop#3312]
Disabling other extensions will still cause this problem.

## To reproduce

1. Create LaTeX file as below.

    ```latex
    \documentclass{article}
    \begin{document}

    \begin{equation}
      f(x) = ax^2 + bx + c
      \label{eqn:quadratic}
    \end{equation}

    \ref{eqn:quadratic}

    \end{document}
    ```

2. Hover the mouse cursor over `\ref{...}` while holding down <kbd>Ctrl</kbd>.
3. `labelNeme` changes color and hover show up. (Click to jump to `\label`)

I think it is generally known how to separate namespaces for label names by category. The behavior of the link being split in two by `:` seems odd.

Not a critical problem, but a bit cluttered.

### Gif animation and screenshots

![refJump](https://user-images.githubusercontent.com/75878492/170726390-0ce06220-81c6-471d-b9e0-d6cb391c979f.gif)

| ![Hover over `eqn`][refJump-12] | ![Hover on `quadratic`][refJump-27] |
| ------------------------------- | ----------------------------------- |

[refJump-12]: https://user-images.githubusercontent.com/75878492/170726382-af9f065b-aa94-4f7b-8686-897ce365d007.png
[refJump-27]: https://user-images.githubusercontent.com/75878492/170726386-ab61be87-f8f7-44a4-81b8-a4a256772c12.png

## Logs

<details>
<summary>
LaTeX Workshop Output
</summary>
<div>

```
[23:55:22] Initializing LaTeX Workshop.
[23:55:22] Extension root: c:\Users\■■■\.vscode\extensions\james-yu.latex-workshop-8.26.0
[23:55:22] $PATH: C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Program Files\Git\usr\bin;C:\Program Files\dotnet\;C:\bin\nodejs\;D:\node_global;C:\Users\■■■\AppData\Local\Microsoft\WindowsApps;C:\bin\VSCode-x64\bin;C:\MinGW\bin;D:\texlive\2021\bin\win32;C:\Users\■■■\AppData\Roaming\npm;D:\bin\Microsoft VS Code Insiders\bin
[23:55:22] $SHELL: undefined
[23:55:22] $LANG: undefined
[23:55:22] $LC_ALL: undefined
[23:55:22] process.platform: win32
[23:55:22] process.arch: x64
[23:55:22] vscode.env.appName: Visual Studio Code
[23:55:22] vscode.env.remoteName: undefined
[23:55:22] vscode.env.uiKind: 1
[23:55:22] Workspace for configuration: file:///d:/vsnotes/EDiT-notes/issue
[23:55:22] editor.acceptSuggestionOnEnter: "on"
[23:55:22] latex-workshop.bind.enter.key: true
[23:55:22] latex-workshop.docker.enabled: false
[23:55:22] latex-workshop.docker.image.latex: ""
[23:55:22] latex-workshop.hover.preview.mathjax.extensions: [
 "boldsymbol",
 "mathtools",
 "empheq",
 "unicode"
]
[23:55:22] latex-workshop.intellisense.package.enabled: true
[23:55:22] latex-workshop.intellisense.update.aggressive.enabled: false
[23:55:22] latex-workshop.intellisense.update.delay: 1000
[23:55:22] latex-workshop.latex.autoBuild.run: "never"
[23:55:22] latex-workshop.latex.build.forceRecipeUsage: true
[23:55:22] latex-workshop.latex.outDir: "%DIR%"
[23:55:22] latex-workshop.latex.recipes: [
 {
  "name": "#LuaLaTeX: 1 time test Build",
  "tools": [
   "lualatex"
  ]
 }
]
[23:55:22] latex-workshop.latex.tools: [
 {
  "name": "lualatex",
  "command": "lualatex",
  "args": [
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ]
 }
]
[23:55:22] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[23:55:22] Creating a new file watcher.
[23:55:22] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[23:55:22] Creating PDF file watcher.
[23:55:22] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[23:55:22] Creating Bib file watcher.
[23:55:22] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[23:55:22] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[23:55:22] [Server] Creating LaTeX Workshop http and websocket server.
[23:55:23] LaTeX Workshop initialized.
[23:55:23] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[23:55:23] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":true,"fieldsOrder":["title","author","editor","year","journal","publisher","school","institution"],"firstEntries":["string","xdata"]}
[23:55:23] Current workspace folders: ["file:///d:/vsnotes/EDiT-notes/issue"]
[23:55:23] Found root file from active editor: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex
[23:55:23] Root file changed: from undefined to d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex
[23:55:23] Start to find all dependencies.
[23:55:23] Root file languageId: latex
[23:55:23] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":55978}
[23:55:23] Reset file watcher.
[23:55:23] Parsing a file and its subfiles: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex
[23:55:23] Parse fls file.
[23:55:23] Cannot find fls file: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.fls
[23:55:23] Added to file watcher: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex
[23:55:23] Snippet data loaded.
[23:55:23] [Server] valdOrigin is http://127.0.0.1:55978
[23:55:23] Checking for duplicate labels: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex.
[23:55:25] Manager.fileWatcher.getWatched: {"d:\\vsnotes\\EDiT-notes\\issue\\James-Yu_LaTeX-Workshop\\labelRefJump":["test.tex"]}
[23:55:25] Manager.filesWatched: ["d:\\vsnotes\\EDiT-notes\\issue\\James-Yu_LaTeX-Workshop\\labelRefJump\\test.tex"]
[23:55:25] BibWatcher.bibWatcher.getWatched: {}
[23:55:25] BibWatcher.bibsWatched: []
[23:55:25] PdfWatcher.pdfWatcher.getWatched: {}
[23:55:25] PdfWatcher.pdfsWatched: []
[23:55:25] PdfWatcher.watchedPdfVirtualUris: []
[23:55:25] PdfWatcher.ignoredPdfUris: []
[23:55:31] Current workspace folders: ["file:///d:/vsnotes/EDiT-notes/issue"]
[23:55:31] Found root file from active editor: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex
[23:55:31] Keep using the same root file: d:\vsnotes\EDiT-notes\issue\James-Yu_LaTeX-Workshop\labelRefJump\test.tex
```

</div>
</details>

## Version

- LaTeX Workshop: 8.26.0
- VS Code
    ```
     Version: 1.67.2 (user setup)
      Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
        Date: 2022-05-17T18:15:52.058Z
    Electron: 17.4.1
    Chromium: 98.0.4758.141
     Node.js: 16.13.0
          V8: 9.8.177.13-electron.0
          OS: Windows_NT x64 10.0.19043
    ```

**Comment by jlelong** [comment]
We have no control from the extension side of what is highlighted when `ctrl` + hover. VS Code highlights the word under the cursor and `:` is not a word character in LaTeX. Because the definition of what a word consists in is quite tricky and is widely used in VS Code, I am not sure adding `:` is a good idea.

https://github.com/James-Yu/LaTeX-Workshop/blob/3c6a257c2dad4ccb65b52f63eb16d39dc19eb922/syntax/latex-language-configuration.json#L120-L123

**Comment by Yarakashi-Kikohshi** [comment]
@jlelong 

I don't know if this will help, but Markdown has `wordPattern` that does not include colon.

https://github.com/microsoft/vscode/blob/a054b98fd4536e2fea87dd01745a5d066a65e257/extensions/markdown-basics/language-configuration.json#L55

In go to definition, the link is not separated by a colon on reference side. But, label side link is split.

```markdown
[reference-style links][url:example]

[url:example]: https://www.example.com
```

![markdownRefStyleJump](https://user-images.githubusercontent.com/75878492/170778544-28a343e8-92af-4b08-abe3-719cebc0a91e.gif)

As for the go to definition of Markdown links, it seems to have been suggested by https://github.com/microsoft/vscode/issues/146293.

**Comment by jlelong** [comment]
After some investigation in the markdown extension, it turns out that they actually use a [`DocumentLinkProvider`](https://code.visualstudio.com/api/references/vscode-api#DocumentLinkProvider) to define links in a document. To my understanding, 

- `ctrl`+ click on a link opens it
- `ctrl`+ click anywhere else,  calls _Go to definition_

See https://github.com/microsoft/vscode/issues/81520

In LW, we do not define a `DocumentLinkProvider`, so behaviours cannot be compared. I am afraid we have to stick to word pattern as far as _Go to definition_  is concerned.

**Comment by James-Yu** [comment]
> In LW, we do not define a `DocumentLinkProvider`, so behaviours cannot be compared. I am afraid we have to stick to word pattern as far as _Go to definition_ is concerned.

Does implement a `DocumentLinkProvider` for LW a good idea? I have not checked and am happy to take it if yes 😄

**Comment by jlelong** [comment]
@James-Yu Let me rephrase. I was just saying that the markdown extension does not seem to face the issue of label names split at `:` because we were looking at links and `ctrl` + click on links calls the `DocumentLinkProvider` instead of using the `DefinitionProvider` through _Go to definition_.

In our case, we do need to rely on the `DefinitionProvider`, which uses the word pattern to delimit the label to highlight. It would be a bad idea to try to turn every symbol into a link because it is not and we would play against the logic of VS Code.

Let us go back to the original issue. The question is: is it worth it adding `:` as a word character to ensure that label names are not split at `:` when calling _Go to definition_? I have to think thoroughly of the implications.

**Comment by jlelong** [comment]
I am back on this `:` issue. The `wordPattern` setting plays a major role in snippet and completion auto-triggering. Considering the issue here is only about visual rendering of _activated_ labels (ctrl click works), I think it is not worth it fiddling wit this setting.
